### DISCUSSION TITLE
Update sudoers  files

### DISCUSSION THREAD

**Discussion by thomasfillon** [ltsp/ltsp#869] [General] [answered]
Hi,

I would like to be able to update the '/etc/sudoers.d/' files from time to time in order to update the users (students) rights and permissions during the school year.

I thought it could be done with `lstp initrd` and copying the server `/etc/sudoers.d` in `/etc/ltsp/etc/sudoers.d/` but it doesn't seem to propagate the files on the clients and I must misunderstand something here.

Is there a way to do this ? I see [here](https://github.com/ltsp/ltsp/discussions/796) that it should be possible with post-init scripts but there might be simpler ways to achieve this.

**Comment by alkisg** [comment, marked as answer]
No, LTSP doesn't automatically copy any directories to the target /.
You can do that easily though with an LTSP command such as:

```
PRE_INIT_CLIENTS_CP_ETC="rw cp -au /etc/ltsp/etc/. /etc/"
```

**Comment by thomasfillon** [comment]
Thanks you very much. It works like a charm !

I can now understand and retrieve the information from `ltsp.conf` man page.

For my understanding, could you please tell me why you choose a  `PRE_INIT_x`  rather than a `POST_INIT_x` command here.

Also, unless I am mistaken, "INIT" is not listed as an APPLET . Does it mean that PRE_APPLET and POST_ APPLET hooks can run if APPLET is either an applet name or "INIT" ?

**Comment by alkisg** [comment]
From https://ltsp.org/man/ltsp:

> LTSP clients also have some additional applets, like initrd-bottom, init and login, but they're not runnable by the user.

And, I chose "PRE_INIT" in case the thing that you want to put in /etc will be read or written by some function of `ltsp init`.
In this case that's not going to happen so it makes no difference, you can use POST_INIT too.

**Reply by thomasfillon** [reply]
Thank you @alkisg That is perfectly clear now.

I will try to find some time to put that somewhere in the wiki because that might be helpful to other.
